### DISCUSSION TITLE
Moving saved bookmarks with the project

### DISCUSSION THREAD

**Discussion by tarekahf** [alefragnani/vscode-bookmarks#490] [Q&A] [answered]
I have opened a Folder as a project in vs code as usual and started using Bookmarks and all is good and I can see that the bookmarks are saved even when I close and re-open the folder.

Then, I decided to open the project from the parent folder, of the original project, and now the bookmarks are lost.

How I can export/import the project from the child folder to be active from the parent folder?

And, how the bookmarks can be saved so that they can be active if the project is moved to another machine?

Tarek

**Comment by alefragnani** [comment, marked as answer]
Hi @tarekahf,

The bookmarks are stored per **session**, which is the same approach VS Code itself works. 

When you open a folder, VS Code maintains a **session** for that folder only, and anything you like searching text, opening files, terminal windows, works only for that folder. When you open a parent/child folder, that info is not available, because VS Code creates another **session**.

With that in mind, the bookmarks will not be available when you open a parent/child folder. But, #287 asks for a **Global Bookmark**, which in the end would ignore this **session** storage and make the bookmarks available anytime. Feel free to upvote the issue and add comments if you have any other idea.

About _moving_ the bookmarks with the folder, take a look at `bookmarks.saveBookmarksInProject` setting. It will make the bookmarks be stored in `.vscode/bookmarks.json` file, inside the folder you opened. It would work just fine for you.

Hope this helps

**Comment by tarekahf** [comment]
Thanks a lot. But, I noticed if I close vs code and open it again, the search history is still there, and also the bookmarks. Or, am I mistaken?

**Reply by alefragnani** [reply]
It's not **session** in a sense of **active session** (open window).

It means that it maintains a **separate state** for each Folder.
